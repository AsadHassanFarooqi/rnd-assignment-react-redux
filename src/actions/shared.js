import { setAuthedUser } from "./authedUser";
import { receiveUsers } from "./users";
import { receiveQuestions } from "./questions";
import { getInitialData } from "../utils/api";

import Cat from '../images/cat.png';

const user = {
    sarahedo: {
        id: 'sarahedo',
        password:'password123',
        name: 'Sarah Edo',
        avatarURL: Cat,
        answers: {
          "8xf0y6ziyjabvozdd253nd": 'optionOne',
          "6ni6ok3ym7mf1p33lnez": 'optionOne',
          "am8ehyc8byjqgar0jgpub9": 'optionTwo',
          "loxhs1bqm25b708cmbf3g": 'optionTwo'
        },
        questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9']
      },
}

export function handleInitialData() {
    return (dispatch) => {
        return getInitialData().then(({users, questions}) => {
            dispatch(receiveUsers(users));
            dispatch(receiveQuestions(questions));
        });
    };
}
