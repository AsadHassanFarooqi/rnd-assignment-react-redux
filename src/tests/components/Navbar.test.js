import React from "react";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";

import Navbar from "../../components/Nav";
import {store} from "../../store";
import {setAuthedUser} from "../../actions/authedUser";

describe("Nav", () => {
  it("should render the View", () => {
    store.dispatch(setAuthedUser({id: "sarahedo", password: ""}));

    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar/>
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  it("should display username of logged in user", () => {
    store.dispatch(setAuthedUser({id: "sarahedo", password: ""}));

    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar/>
        </BrowserRouter>
      </Provider>
    );

    const userSpanElement = view.getByTestId("user-information");
    expect(userSpanElement.textContent).toBe("sarahedo");

  });
});
