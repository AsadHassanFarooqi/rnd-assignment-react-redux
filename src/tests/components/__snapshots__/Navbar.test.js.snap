// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render the View 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="px-2 mb-5 navbar navbar-expand navbar-dark bg-dark"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link mx-2"
            href="/add"
          >
            Add
          </a>
          <a
            class="nav-link mx-2"
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <button
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
        </div>
        <div
          class="justify-content-end navbar-collapse collapse"
        >
          <span
            class="navbar-text"
          >
            Logout: 
            <a
              data-testid="user-information"
              href="/"
            >
              sarahedo
            </a>
          </span>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="px-2 mb-5 navbar navbar-expand navbar-dark bg-dark"
    >
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="nav-link"
          href="/"
        >
          Home
        </a>
        <a
          class="nav-link mx-2"
          href="/add"
        >
          Add
        </a>
        <a
          class="nav-link mx-2"
          href="/leaderboard"
        >
          Leaderboard
        </a>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <span
          class="navbar-text"
        >
          Logout: 
          <a
            data-testid="user-information"
            href="/"
          >
            sarahedo
          </a>
        </span>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
